### PR TITLE
Use rules_jvm_external

### PR DESCRIPTION
**Pull request by Kernald**
This replaces the previous in-repo Maven fetcher, and removes the dependency on the Android rules - which seemed to be unused entirely?

This also depends on #86 to be merged, and would render #60 obsolete.

### CODE REVIEW

**Comment by Kernald** [comment]
Looks like this needs Bazel 0.25+ to run. Should I update the version used by Travis?

**Comment by achew22** [comment]
@kernald, have you considered getting into the [Bazel team's CI system](https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/README.md)?

**Comment by Kernald** [comment]
That definitely looks interesting, I'm in no way responsible for this repository though :-)

**Comment by achew22** [comment]
Ah, then to the maintainers of this repo. If you would like to have a TAP like system (essentially no config and it works great) for your github, all you have to do is follow the linked instructions and the Bazel team will run your CI infra for you.

**Comment by cpovirk** [comment]
Thanks!

> Looks like this needs Bazel 0.25+ to run. Should I update the version used by Travis?

Yeah, let's do it. I guess we can go all the way to 0.29.1?

**Comment by cpovirk** [comment]
> @Kernald, have you considered getting into the [Bazel team's CI system](https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/README.md)?

Thanks, we should look at that. I've filed https://github.com/google/bazel-common/issues/88

**Comment by Kernald** [comment]
> Yeah, let's do it. I guess we can go all the way to 0.29.1?

Done in #89

File: workspace_defs.bzl [hunk 1]
```diff
@@ -47,221 +31,80 @@ def google_common_workspace_rules():
...
+    GRPC_VERSION = "1.2.0"
+    LOG4J2_VERSION = "2.11.2"
+    INCAP_VERSION = "0.2"
 
-    _maven_import(
-        artifact = "com.google.protobuf:protobuf-java:3.7.0",
-        licenses = ["notice"],
-        sha256 = "dc7f93e3a3dc2c11be5ba9672af3e26410f0a3289312dbf2260d4d8a0c711a51",
+    maven_install(
+        artifacts = [
```
Comment (jin) [inline workspace_defs.bzl]: I recommend making this list available as a variable for downstream consumers to import into their local maven_install instance. See https://github.com/bazelbuild/rules_jvm_external/#exporting-and-consuming-artifacts-from-external-repositories

Comment (Kernald) [reply workspace_defs.bzl]: Does that mean that we should only call `maven_install` in the `WORKSPACE` and rely on every user to call `maven_install` with both their dependencies and this list?

Comment (jin) [reply workspace_defs.bzl]: Yeah, this has the advantage of ensuring that's only one version of every artifact in a single namespace (`@maven`). I'd think that most bazel-common projects would also need to pull other maven artifacts as well.

Do you think this is a good user experience?

Comment (Kernald) [reply workspace_defs.bzl]: While I definitely see the value into having a single version of the dependencies, this requires downstream consumer to have a call to `maven_install`, while the purpose of bazel-common seems to be to minimise the set-up required in downstream workspaces. Maybe @ronshapiro would have an input on that?

It seems that we have two alternatives: either downstream consumers have to call `maven_install` and pass in bazel-common's list of artifacts, or we can call `maven_install` in bazel-common's set-up with a specific repository name, downstream consumers don't need to do anything, but if they use some Maven dependencies not provided by bazel-common, they're likely to end up with different versions of transitive dependencies that could have been provided by bazel-common (which is the current behaviour, as far as I can tell).

Comment (jin) [reply workspace_defs.bzl]: Yeah, I'd prefer the former, but the latter works as well. I'll leave that up to the maintainers of this repo to decide.

**Pull request closed without merging**